test/extended/cli/template.go

```diff
@@ -16,10 +18,299 @@ var _ = g.Describe("[sig-cli] templates", func() {
+		err := oc.Run("get").Args("templates").Execute()
+		o.Expect(err).NotTo(o.HaveOccurred())
+		err = oc.Run("create").Args("-f", appTemplatePath).Execute()
+		o.Expect(err).NotTo(o.HaveOccurred())
+		err = oc.Run("get").Args("templates").Execute()
+		o.Expect(err).NotTo(o.HaveOccurred())
+		err = oc.Run("get").Args("templates", "ruby-helloworld-sample").Execute()
+		o.Expect(err).NotTo(o.HaveOccurred())
+		outputYamlFile, err := oc.WithoutNamespace().Run("get").Args("template", "ruby-helloworld-sample", "-o", "json").OutputToFile("template.json")
+		o.Expect(err).NotTo(o.HaveOccurred())
+		err = oc.WithoutNamespace().Run("process").Args("-f", outputYamlFile).Execute()
+		o.Expect(err).NotTo(o.HaveOccurred())
+		err = oc.Run("process").Args("ruby-helloworld-sample").Execute()
+		o.Expect(err).NotTo(o.HaveOccurred())
+		out, err := oc.Run("process").Args("ruby-helloworld-sample", "-o", "template", "--template", "{{.kind}}").Output()
+		o.Expect(err).NotTo(o.HaveOccurred())
+		o.Expect(out).To(o.Equal("List"))
+		out, err = oc.Run("process").Args("ruby-helloworld-sample", "-o", "go-template", "--template", "{{.kind}}").Output()
+		o.Expect(err).NotTo(o.HaveOccurred())
+		o.Expect(out).To(o.Equal("List"))
+		out, err = oc.Run("process").Args("ruby-helloworld-sample", "-o", "go-template={{.kind}}").Output()
+		o.Expect(err).NotTo(o.HaveOccurred())
+		o.Expect(out).To(o.Equal("List"))
+		err = oc.Run("process").Args("ruby-helloworld-sample", "-o", "go-template-file=/dev/null").Execute()
+		o.Expect(err).NotTo(o.HaveOccurred())
+		out, err = oc.Run("process").Args("ruby-helloworld-sample", "-o", "jsonpath", "--template", "{.kind}").Output()
+		o.Expect(err).NotTo(o.HaveOccurred())
+		o.Expect(out).To(o.Equal("List"))
+		out, err = oc.Run("process").Args("ruby-helloworld-sample", "-o", "jsonpath={.kind}").Output()
+		o.Expect(err).NotTo(o.HaveOccurred())
+		o.Expect(out).To(o.Equal("List"))
+		err = oc.Run("process").Args("ruby-helloworld-sample", "-o", "jsonpath-file=/dev/null").Execute()
+		o.Expect(err).NotTo(o.HaveOccurred())
+		out, err = oc.Run("process").Args("ruby-helloworld-sample", "-o", "describe").Output()
+		o.Expect(err).NotTo(o.HaveOccurred())
+		o.Expect(out).To(o.ContainSubstring("ruby-27-centos7"))
+		out, err = oc.Run("process").Args("ruby-helloworld-sample", "-o", "json").Output()
+		o.Expect(err).NotTo(o.HaveOccurred())
+		o.Expect(out).To(o.ContainSubstring("ruby-27-centos7"))
+		out, err = oc.Run("process").Args("ruby-helloworld-sample", "-o", "yaml").Output()
+		o.Expect(err).NotTo(o.HaveOccurred())
+		o.Expect(out).To(o.ContainSubstring("ruby-27-centos7"))
+		out, err = oc.Run("process").Args("ruby-helloworld-sample", "-o", "name").Output()
+		o.Expect(err).NotTo(o.HaveOccurred())
+		o.Expect(out).To(o.ContainSubstring("ruby-27-centos7"))
+		out, err = oc.Run("describe").Args("templates", "ruby-helloworld-sample").Output()
+		o.Expect(err).NotTo(o.HaveOccurred())
+		o.Expect(out).To(o.ContainSubstring("ruby-sample-build"))
```

Comment:
Questionable whether `o.Expect(out).To(o.ContainSubstring("BuildConfig"))` should be checked as well to stay aligned with the original `os::cmd::expect_success_and_text 'oc describe templates ruby-helloworld-sample' "BuildConfig.*ruby-sample-build"` line.